app/components/curriculum-inventory-report-header.js

```diff
@@ -17,16 +20,54 @@ const Validations = buildValidations({
 });
 
 export default Component.extend(Validations, ValidationErrorDisplay, {
+  flashMessages: service(),
+
   didReceiveAttrs(){
     this._super(...arguments);
     this.set('reportName', this.get('report.name'));
     this.set('isFinalized', this.get('report.isFinalized'));
+
+    // check if the d/l link points at the same domain as the current page is on.
+    let parser = document.createElement('a');
+    parser.href = this.get('report.absoluteFileUri');
+    let backendDomain = parser.hostname;
+    let frontendDomain = window.location.hostname;
+    this.set('downloadFromSameDomain', (backendDomain === frontendDomain));
```

Comment:
i don't think that's necessary in this particular situation. `didReceiveAttrs` _will_ re-fire each time `report` changes.

"Since the didReceiveAttrs hook is called every time a component's attributes are updated whether on render or re-render, you can use the hook to effectively act as an observer, ensuring code is executed every time an attribute changes."

https://guides.emberjs.com/v2.6.0/components/the-component-lifecycle/#toc_formatting-component-attributes-with-code-didreceiveattrs-code